src/native/minipal/utf8.h

```diff
@@ -23,25 +24,57 @@ typedef wchar_t CHAR16_T;
 typedef unsigned short CHAR16_T;
 #endif
 
-int minipal_utf8_to_utf16_preallocated(const char* source, size_t sourceLength, CHAR16_T** destination, size_t destinationLength, unsigned int flags
-#if BIGENDIAN
-    , bool treatAsLE
-#endif
-);
+/**
+ * Convert a string from UTF-8 to UTF-16 (UCS-2) with preallocated memory
+ *
+ * @param source The source string in UTF-8 format.
+ * @param sourceLength Length of the source string.
+ * @param destination Pointer to the destination UTF-16 string. It can be NULL to query number of items required by the conversion.
+ * @param destinationLength Length of the destination string.
+ * @param flags Flags to alter the behavior of converter. Supported flags are MINIPAL_MB_ERR_INVALID_CHARS and MINIPAL_TREAT_AS_LITTLE_ENDIAN.
+ * @return Number of items required by the conversion.
```

Comment:
If the input size is `size_t`, should the output size be `size_t` as well?

Reply:
Sure